### PR TITLE
Bcol staff payment

### PR DESCRIPTION
**Pull request by sumesh-aot**
*Issue #:*
https://github.com/bcgov/entity/issues/4235

*Description of changes:*
- Changes to bcol-api to use applied charge web service for staff payments
- Changes to pay-api to send in datNumber and bcolAccountNumber for staff payments
- Changes for staff fee code for BCOL staff payments. 
- Fixing free filing service fee bug 
- Adding DAT number to receipt

*Checklist:*
I confirm that below checklist items are addressed in this pull request; (check the boxes applicable)
- [ ] No lint errors
- [ ] No test case failures and proper coverage for all modules/classes
- [ ] Updated the deployment configs for new environment variable(s)
- [ ] Updated the postman collection in entity repository (https://github.com/bcgov/entity/tree/master/api-e2e/postman) for e2e tests

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the sbc-pay license (Apache 2.0).

### CODE REVIEW

File: .github/workflows/bcol-api-check.yml [hunk 1]
```diff
@@ -27,6 +27,8 @@ jobs:
       BCOL_DEBIT_ACCOUNT_VERSION: "fake"
       BCOL_LINK_CODE: "000"
 
+      BCOL_APPLIED_CHARGE_WSDL_URL: "https://appsd1.bconline.gov.bc.ca/account/services/AppliedCharge?wsdl"
```
Comment (pwei1018) [inline .github/workflows/bcol-api-check.yml:30]: Is possible using mock endpoint?

Comment (sumesh-aot) [reply .github/workflows/bcol-api-check.yml:30]: test doesn't actually hit the actual soap service, all are using mocked response. A valid wsdl is needed for the app to parse. we would need a mock soap wsdl url.

File: bcol-api/src/bcol_api/resources/bcol_payment.py [hunk 2]
```diff
@@ -42,6 +43,13 @@ class AccountPayment(Resource):
     def post():
         """Create a payment record in BCOL."""
         try:
+            if _jwt.validate_roles([Role.STAFF.value, Role.EDIT.value]):
```
Comment (saravanpa-aot) [inline bcol-api/src/bcol_api/resources/bcol_payment.py:46]: cursios , why _jwt ;not just jwt without underscore?

File: pay-api/migrations/versions/cf9a60955b68_bcol_staff_payment.py [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+down_revision = 'e8edc889072d'
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    # ### commands auto generated by Alembic - please adjust! ###
+    op.add_column('corp_type', sa.Column('bcol_staff_fee_code', sa.String(length=20), nullable=True))
+    op.add_column('invoice', sa.Column('dat_number', sa.String(length=50), nullable=True))
+    op.create_index(op.f('ix_invoice_dat_number'), 'invoice', ['dat_number'], unique=False)
```
Comment (saravanpa-aot) [inline pay-api/migrations/versions/cf9a60955b68_bcol_staff_payment.py:23]: i hope this table doesnt already have a lot of indexes

File: report-api/report-templates/payment_receipt.html [hunk 4]
```diff
@@ -419,19 +419,19 @@
...
             </div>
             {% endif %}
-            {% if invoice.get('folioNumber', 'None') != 'None' %}
+            {% if invoice.get('folioNumber', None) %}
             <div class="payment-row">
               Folio Number: <span> {{invoice['folioNumber']}}</span>
             </div>
             {% endif %}
-            {% if routingSlipNumber %}
+            {% if invoice.get('routingSlip', None) %}
```
Comment (saravanpa-aot) [inline report-api/report-templates/payment_receipt.html:427]: i forgot .just making sure... does this change need no change in pay-api?

File: bcol-api/src/bcol_api/resources/bcol_payment.py (hunk 2 above)
Comment (sumesh-aot) [reply bcol-api/src/bcol_api/resources/bcol_payment.py:46]: We started using the _ for this decorators, as this instance won't be passed to any other class/function.

File: pay-api/migrations/versions/cf9a60955b68_bcol_staff_payment.py (hunk 3 above)
Comment (sumesh-aot) [reply pay-api/migrations/versions/cf9a60955b68_bcol_staff_payment.py:23]: Invoice have some indexes for most of the columns which are searchable.. Yea, we may need to refactor later if there is any performance impacts.

File: report-api/report-templates/payment_receipt.html (hunk 4 above)
Comment (sumesh-aot) [reply report-api/report-templates/payment_receipt.html:427]: Pay already passes this, as it passes the whole invoice dict to report-api.

File: pay-api/migrations/versions/cf9a60955b68_bcol_staff_payment.py (hunk 3 above)
Comment (saravanpa-aot) [reply pay-api/migrations/versions/cf9a60955b68_bcol_staff_payment.py:23]: okay..

**Comment by sumesh-aot** [comment]
QA will need to coordinate this with Entities QA team

**Pull request merged**